### ISSUE TITLE
`no-invalid-triple-slash-reference` rule incorrectly warns on `path` triple slash references in `.js` files

### ISSUE THREAD

**Issue by jaydenseric** [denoland/deno_lint#982]
### Lint Name

`no-invalid-triple-slash-reference`

### Code Snippet

```js
// @ts-check
/// <reference path="./types.d.ts" />
```

### Expected Result

No lint errors.

### Actual Result

Lint errors:

```
This triple-slash reference directive is not allowed in JavaScript
In JavaScript only the `lib` and `types` directives are allowed, like `/// <reference lib="..." />` or `/// <reference types="..." />`
```

### Additional Info

A similar bug was already fixed for `lib` triple slash references: https://github.com/denoland/deno_lint/issues/854 .

### Version

deno 1.16.4 (release, aarch64-apple-darwin)
v8 9.7.106.15
typescript 4.4.2

**Comment by AkinAguda** [comment]
Can I give this a shot?

**Comment by bartlomieju** [comment]
@AkinAguda please do
